### ISSUE TITLE
Expand Dioptra Client to cover existing REST API

### ISSUE THREAD

**Issue by hbooth** [usnistgov/dioptra#211]
The current Dioptra Client only covers a subset of the existing REST API and should use composition with each top-level API resource being a composed class within the client code base.

## Design Spec

The formal client object is built using composition. Here's a quick spec on how that will look:

```python
class DioptraClient(object):
    def __init__(self, experiment, job, queue, ...):
        self._experiment = experiment
        self._job = job
        self._queue = queue
        # Set other endpoint objects in the same manner

    @property
    def experiment(self):
        return self._experiment

    @property
    def job(self):
        return self._job

    @property
    def queue(self):
        return self._queue
```

Using the client to submit a job would then work like this:

```python
client.job.submit(...)
```

Each sub-client in the main client represents an endpoint namespace. The classes for these endpoint clients will look something like this:

```python
class JobClient(object):
    def get_all(self) -> list[dict[str, Any]]:
        # Body of function follows

    def submit(
        self,
        workflows_file: str | Path,
        experiment_name: str,
        entry_point: str,
        entry_point_kwargs: str | None = None,
        depends_on: str | None = None,
        queue: str = "tensorflow_cpu",
        timeout: str = "24h",
    ) -> dict[str, Any]:
    # Body of function follows

# Classes like ExperimentClient, QueueClient follow
```

The function used to perform dependency injection and instantiate `DioptraClient` would look something like this:

```python
def get_dioptra_client(address: str | None = None, ...) -> DioptraClient:
    address = (
        f"{address}/api" if address else f"{os.environ['DIOPTRA_RESTAPI_URI']}/api"
    )
    scheme, netloc, path, _, _, _ = urlparse(address)
    # More needed with the address most likely

    experiment_client = ExperimentClient(...)
    job_client = JobClient(...)
    queue_client = QueueClient(...)

    return DioptraClient(
        experiment=experiment_client,
        job=job_client,
        queue=queue_client,
        ...
    )
```

In addition, it will likely be necessary to have a requests session set up to be able to handle and manage the use of cookies and other header information while using the client, so after the initial port, investigate switching to using a session object instead of direct calls to `requests.get`, etc.

## Depends On

- #210 

## Definition of Done

- [ ] Add new classes for each resource
- [ ] Class for each Resource covers all REST API calls
- [ ] New process to instantiate the new client use dependency-injection
- [ ] Merge into Dev

**Comment by keithmanville** [comment]
See closed PR #247 and the `abyrne-refactor-client` branch.
Will revisit the python client once V1 of the REST API is complete.

**Comment by keithmanville** [comment]
stale
